NewForce-Cohort-9/newforce-c9-tabloid-full-stack-TabloidFullStack
# #20 Create A Comment

### Description

Users can now create new comments.

Given the user is viewing the details of a post, when they select the "Add Comment" menu option they should be directed to a page with a form for creating a new comment. Given the user has entered the relevant information for a comment, when they click the "Save" button, then the Comment should be saved to the database, and the creation datetime should be automatically set to the current date and time, and the user should be redirected to the new comments list page for the related post.

[Trello Ticket #20 Create A Comment](https://trello.com/c/axMqk9kV)

### Type of Change


### Testing Instructions

1. `git add` and `git commit` whatever you're working on on your branch
2. `git checkout main`
3. `git fetch --a`
4. `git checkout <js-AddComment>` 
5. If you don't see the most recent version of my files, `git pull origin <js-AddComment>`
6. If you haven't created the Tabloid database, please do so.
7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-api-all-stars/blob/main/SQL/01_Tabloid_Create_DB.sql)
8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-api-all-stars/blob/main/SQL/01_Tabloid_Create_DB.sql)
9. Click the execute with debugger solid green button in purple Visual Studio to launch the api
10. In your terminal, launch the program within the root folder with the command `npm start`
11. In browser, please go to `Login` on the nav bar and log in with this email: `foo@bar.com`
12. This should redirect you back to the home page. 
13. You should  see `Posts` on the nav bar.
14. Select `Posts` to be directed to `/posts` where you should see a list of all posts.
15. Select a post to enter post details for that post
16. Select "View Comments" to view a list of comments that are associated with that post in order from newest to oldest.
17. Select "Add New Comment" to be redirected to a form prompting you to fill out the Subject and Content of a new comment. Please make a new Comment with the subject "Test" and the Content "Test Content".
18. Click the "Submit" button to be redirected back to the Comment list. The Test Comment should be at the top of the list.

### Checklist:
